components/camel-spring/src/main/java/org/apache/camel/spring/SpringCamelContext.java

```diff
@@ -159,7 +159,7 @@ public void stop() {
     public void onApplicationEvent(ApplicationEvent event) {
         LOG.debug("onApplicationEvent: {}", event);
 
-        if (event instanceof ContextRefreshedEvent) {
+        if (event instanceof ContextRefreshedEvent && ((ContextRefreshedEvent) event).getApplicationContext().getId().equals(this.applicationContext.getId())) {
```

Comment:
We should watch out for `ApplicationContext::getId` returning `null`
```suggestion
        if (event instanceof ContextRefreshedEvent && ((ContextRefreshedEvent) event).getApplicationContext() == this.applicationContext) {
```